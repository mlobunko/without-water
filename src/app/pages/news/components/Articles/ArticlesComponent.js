import React from 'react';
import PropTypes from 'prop-types';

import CounterNews from './CounterNews';
import { StyledJumbotron } from './styles';
import { StyledArticlesSection, StyledHeader } from './styles';

export const ArticlesComponent = ({ news }) => (
  <React.Fragment>
    <StyledArticlesSection>
      {news.map(({ title, text }) => (
        <StyledJumbotron key={title}>
          <StyledHeader>{title}</StyledHeader>
          <p>{text}</p>
        </StyledJumbotron>
      ))}
    </StyledArticlesSection>
    <CounterNews total={news.length} />
  </React.Fragment>
);
ArticlesComponent.displayName = 'ArticlesComponent';

export default ArticlesComponent;

ArticlesComponent.propTypes = {
  news: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired
    })
  )
};
