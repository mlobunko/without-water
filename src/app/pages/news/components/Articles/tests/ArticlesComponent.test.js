import React from 'react';
import { shallow } from 'enzyme';

import { ArticlesComponent } from '../ArticlesComponent';

describe('render', () => {
  it('renders correctly', () => {
    const news = [{ title: '123', text: 'hola' }];
    const wrapper = shallow(<ArticlesComponent news={news} />);
    expect(wrapper).toMatchSnapshot();
  });
});
