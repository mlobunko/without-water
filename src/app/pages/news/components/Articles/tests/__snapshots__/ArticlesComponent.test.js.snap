// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly 1`] = `
<Fragment>
  <styled.div>
    <Styled(Jumbotron)
      key="123"
    >
      <styled.h3>
        123
      </styled.h3>
      <p>
        hola
      </p>
    </Styled(Jumbotron)>
  </styled.div>
  <Component
    total={1}
  />
</Fragment>
`;
